python/servo/build_commands.py

```diff
@@ -64,41 +56,14 @@ class MachCommands(CommandBase):
                      help='Print very verbose output')
     @CommandArgument('params', nargs='...',
                      help="Command-line arguments to be passed through to Cargo")
-    @CommandBase.build_like_command_arguments
-    def build(self, release=False, dev=False, jobs=None, params=None, no_package=False,
+    @CommandBase.common_command_arguments(build_configuration=True, build_type=True)
+    def build(self, build_type: BuildType, jobs=None, params=None, no_package=False,
               verbose=False, very_verbose=False, libsimpleservo=False, **kwargs):
         opts = params or []
         has_media_stack = "media-gstreamer" in self.features
 
-        release_path = path.join(self.target_path, "release", "servo")
-        dev_path = path.join(self.target_path, "debug", "servo")
-
-        release_exists = path.exists(release_path)
-        dev_exists = path.exists(dev_path)
-
-        if not (release or dev):
-            if self.config["build"]["mode"] == "dev":
```

Comment:
I believe `self.config["build"]["mode"]` is what the user has specified in their `.servobuild` file. Wouldn't removing this code also break support for specifying build mode via `.servobuild`?

Reply:
Hrm. That's a good point. I did end up removing this functionality, but probably this should be removed more thoroughly if I'm removing it here.

My thought is that we should remove this setting entirely. One day hopefully, when you pass --dev or --release to cargo this will work like passing the same option to mach. Instead of configuring how mach builds, you will configure how cargo builds.

Reply:
I've gone ahead and removed the build mode configuration from the `servobuild.example` file.

Reply:
FWIW, I was thinking we should retain support until we remove the concept of `.servobuild` completely. My main concern is that it seems counterintuitive to support `.servobuild` but not allow complete build configuration to be specified via the file, but I guess .servobuild is not used that much in practice.

Reply:
I've gone ahead and restored the `.servobuild` functionality for selecting the default build type. This is now printed a well.